packages/workflow/src/internals.ts

```diff
@@ -664,6 +754,30 @@ export class Activator implements ActivationHandler {
     });
   }
 
+  private acceptUpdate(updateId: string): void {
+    const protocolInstanceId = updateId;
+    this.pushCommand({
+      updateResponse: { protocolInstanceId, accepted: this.payloadConverter.toPayload(null) },
+    });
+  }
+
+  private completeUpdate(updateId: string, result: unknown): void {
+    const protocolInstanceId = updateId;
+    this.pushCommand({
+      updateResponse: { protocolInstanceId, completed: this.payloadConverter.toPayload(result) },
+    });
+  }
+
+  private rejectUpdate(updateId: string, error: unknown): void {
+    const protocolInstanceId = updateId;
+    this.pushCommand({
+      updateResponse: {
+        protocolInstanceId,
+        rejected: this.errorToFailure(ensureTemporalFailure(error)),
+      },
+    });
+  }
```

Comment:
nit: Feels like overkill to me, I'd inline these methods.

Reply:
> nit: Feels like overkill to me, I'd inline these methods.

I don't agree with this; I'm fairly strongly in the functions-should-always-be-short camp. The helper functions here were deliberately broken out with clear names to make the main implementation short and have a declarative style. In addition to that being a general programming style opinion that I hold fairly firmly, this particular part of the code (the execution of the user validator and update handler) is responsible for upholding a few fairly important guarantees (fail task vs fail update, handler must not see mutations to args made by validator, handler scheduling considerations), and it's been tricky to get it both correct and readable across our SDKs, so there's an even stronger argument for leaning toward the short-and-declarative over the long-and-procedural style here. Below's a comparison of what it looks like with them inlined. There are worse examples of long-and-procedural code, but the high-level algorithm is significantly harder to discern and it has become much harder for a reader to pick out the important fact that we are rejecting in two places, accepting in one and completing in one. 


<img width="1551" alt="image" src="https://github.com/temporalio/sdk-typescript/assets/52205/ce4e9609-f1b2-4222-926f-4d7554537384">